File: src/TuRuta/TuRuta.Client/Routes/RoutesClient.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

using TuRuta.Common.ViewModels;

namespace TuRuta.Client.Routes
{
    public class RoutesClient
    {
        public RoutesClient(string baseAddress = null)
        {
            var address = baseAddress ?? "http://localhost:56340/";
            HttpClient.BaseAddress = new Uri(address);
        }

        private HttpClient HttpClient { get; } = new HttpClient();

        public async Task<IEnumerable<string>> Find(string hint)
        {
            var response = await HttpClient.GetAsync($"/api/routes/find/{hint}");
            if (response.IsSuccessStatusCode)
            {
                return JsonConvert.DeserializeObject<IEnumerable<string>>(await response.Content.ReadAsStringAsync());
            }

            return default(IEnumerable<string>);
        }

        public async Task<RouteVM> Get(string name)
        {
            var response = await HttpClient.GetAsync($"/api/routes/{name}");
            if (response.IsSuccessStatusCode)
            {
                return JsonConvert.DeserializeObject<RouteVM>(await response.Content.ReadAsStringAsync());
            }

            return default(RouteVM);
        }
    }
}

File: src/TuRuta/TuRuta.Web/ClientApp/clients/RoutesClient.ts
export default class RoutesClient {
    async AddStops(routeId: string, stopIds: string[]) {
        var response = await fetch('api/routes/addstops/' + routeId, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(stopIds)
        });
        if (response.ok) {
            var json = await response.json();
            return json as routeVM;
        }

        return null;
    }

    async AddStop(routeId:string, stopId: string) {
        var response = await fetch('api/routes/addstop/'+ routeId + '/' + stopId);
        if (response.ok) {
            var json = await response.json();
            return json as routeVM;
        }

        return null;
    }

    async Search(hint: string) {
        var response = await fetch('api/routes/find/' +  hint);
        if (response.ok) {
            var json = await response.json()
            return json as string[];
        }

        return null;
    }

    async Names() {
        var response = await fetch("api/routes/names");
        if (response.ok) {
            var json = await response.json();
            return json as string[];
        }

        return null;
    }

    async Create(name: string) {
        var response = await fetch('api/routes/create/' + name);
        if (response.ok) {
            var json = await response.json();
            return json as routeVM;
        }

        return null;
    }

    async Get(name: string) {
        var response = await fetch('api/routes/' + name);
        if (response.ok) {
            var json = await response.json();
            return json as routeVM;
        }

        return null;
    }
}